lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain_ports.py

```
@@ -0,0 +1,237 @@
+        :return: Details about the broadcast domain. None if not found.
+        :rtype: dict
+        """
+        domain_get_iter = netapp_utils.zapi.NaElement('net-port-broadcast-domain-get-iter')
+        broadcast_domain_info = netapp_utils.zapi.NaElement('net-port-broadcast-domain-info')
+        broadcast_domain_info.add_new_child('broadcast-domain', self.broadcast_domain)
+        query = netapp_utils.zapi.NaElement('query')
+        query.add_child_elem(broadcast_domain_info)
+        domain_get_iter.add_child_elem(query)
+        result = self.server.invoke_successfully(domain_get_iter, True)
+        domain_exists = None
+        # check if broadcast domain exists
+        if result.get_child_by_name('num-records') and \
+                int(result.get_child_content('num-records')) == 1:
+            domain_info = result.get_child_by_name('attributes-list').get_child_by_name('net-port-broadcast-domain-info')
+            domain_name = domain_info.get_child_content('broadcast-domain')
+            domain_ports = domain_info.get_child_content('port-info')
+            domain_exists = {
+                'domain-name': domain_name,
+                'ports': domain_ports
+            }
+        return domain_exists
+
+    def create_broadcast_domain_ports(self):
+        """
+        Creates new broadcast domain ports
+        """
+        domain_obj = netapp_utils.zapi.NaElement('net-port-broadcast-domain-add-ports')
+        domain_obj.add_new_child("broadcast-domain", self.broadcast_domain)
+        if self.ipspace:
+            domain_obj.add_new_child("ipspace", self.ipspace)
+        if self.ports:
+            ports_obj = netapp_utils.zapi.NaElement('ports')
+            domain_obj.add_child_elem(ports_obj)
+            for port in self.ports:
+                ports_obj.add_new_child('net-qualified-port-name', port)
+        try:
+            self.server.invoke_successfully(domain_obj, True)
+            return True
```

This is also not properly idempotent- you're not comparing to the existing port list, so always returning "changed: True" even if it's already in the right state.

---

If the port given is already in use our zapi will fail with error 18605, so we return false in that case. 
        except netapp_utils.zapi.NaApiError as error:
            # Error 18605 denotes port already being used.
            if to_native(error.code) == "18605":
                return False

---

Is that error "the port is in use by something" or "port is already in the state you asked for"? If it's the former (as I suspect), it's arguably incorrect to silently succeed (as the port will not be in the requested state since it's a member of the wrong broadcast domain).